{
    "dataType": "CVE_RECORD",
    "dataVersion": "5.1",
    "cveMetadata": {
        "cveId": "CVE-2024-34073",
        "assignerOrgId": "a0819718-46f1-4df5-94e2-005712e83aaa",
        "state": "PUBLISHED",
        "assignerShortName": "GitHub_M",
        "dateReserved": "2024-04-30T06:56:33.382Z",
        "datePublished": "2024-05-03T10:11:12.203Z",
        "dateUpdated": "2024-07-05T16:39:38.270Z"
    },
    "containers": {
        "cna": {
            "title": "Command Injection in sagemaker-python-sdk",
            "problemTypes": [
                {
                    "descriptions": [
                        {
                            "cweId": "CWE-78",
                            "lang": "en",
                            "description": "CWE-78: Improper Neutralization of Special Elements used in an OS Command ('OS Command Injection')",
                            "type": "CWE"
                        }
                    ]
                }
            ],
            "metrics": [
                {
                    "cvssV3_1": {
                        "attackComplexity": "LOW",
                        "attackVector": "LOCAL",
                        "availabilityImpact": "HIGH",
                        "baseScore": 7.8,
                        "baseSeverity": "HIGH",
                        "confidentialityImpact": "HIGH",
                        "integrityImpact": "HIGH",
                        "privilegesRequired": "NONE",
                        "scope": "UNCHANGED",
                        "userInteraction": "REQUIRED",
                        "vectorString": "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
                        "version": "3.1"
                    }
                }
            ],
            "references": [
                {
                    "name": "https://github.com/aws/sagemaker-python-sdk/security/advisories/GHSA-7pc3-pr3q-58vg",
                    "tags": [
                        "x_refsource_CONFIRM"
                    ],
                    "url": "https://github.com/aws/sagemaker-python-sdk/security/advisories/GHSA-7pc3-pr3q-58vg"
                },
                {
                    "name": "https://github.com/aws/sagemaker-python-sdk/pull/4556",
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://github.com/aws/sagemaker-python-sdk/pull/4556"
                },
                {
                    "name": "https://github.com/aws/sagemaker-python-sdk/commit/2d873d53f708ea570fc2e2a6974f8c3097fe9df5",
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://github.com/aws/sagemaker-python-sdk/commit/2d873d53f708ea570fc2e2a6974f8c3097fe9df5"
                }
            ],
            "affected": [
                {
                    "vendor": "aws",
                    "product": "sagemaker-python-sdk",
                    "versions": [
                        {
                            "version": "< 2.214.3",
                            "status": "affected"
                        }
                    ]
                }
            ],
            "providerMetadata": {
                "orgId": "a0819718-46f1-4df5-94e2-005712e83aaa",
                "shortName": "GitHub_M",
                "dateUpdated": "2024-05-03T10:11:12.203Z"
            },
            "descriptions": [
                {
                    "lang": "en",
                    "value": "sagemaker-python-sdk is a library for training and deploying machine learning models on Amazon SageMaker. In affected versions the capture_dependencies function in `sagemaker.serve.save_retrive.version_1_0_0.save.utils` module allows for potentially unsafe Operating System (OS) Command Injection if inappropriate command is passed as the “requirements_path” parameter. This consequently may allow an unprivileged third party to cause remote code execution, denial of service, affecting both confidentiality and integrity. This issue has been addressed in version 2.214.3. Users are advised to upgrade. Users unable to upgrade should not override the “requirements_path” parameter of capture_dependencies function in `sagemaker.serve.save_retrive.version_1_0_0.save.utils`, and instead use the default value."
                }
            ],
            "source": {
                "advisory": "GHSA-7pc3-pr3q-58vg",
                "discovery": "UNKNOWN"
            }
        },
        "adp": [
            {
                "providerMetadata": {
                    "orgId": "96f1cf3c-f80a-4e0f-a063-99a91e3f1325",
                    "shortName": "secretariat-reference",
                    "dateUpdated": "2024-07-05T16:39:38.270Z"
                },
                "references": [
                    {
                        "name": "https://github.com/aws/sagemaker-python-sdk/security/advisories/GHSA-7pc3-pr3q-58vg",
                        "tags": [
                            "x_refsource_CONFIRM",
                            "x_cve_prog_secretariat_snapshot"
                        ],
                        "url": "https://github.com/aws/sagemaker-python-sdk/security/advisories/GHSA-7pc3-pr3q-58vg/DEMO_2024_06_19"
                    },
                    {
                        "name": "https://github.com/aws/sagemaker-python-sdk/pull/4556",
                        "tags": [
                            "x_refsource_MISC",
                            "x_cve_prog_secretariat_snapshot"
                        ],
                        "url": "https://github.com/aws/sagemaker-python-sdk/pull/4556/DEMO_2024_06_19"
                    },
                    {
                        "name": "https://github.com/aws/sagemaker-python-sdk/commit/2d873d53f708ea570fc2e2a6974f8c3097fe9df5",
                        "tags": [
                            "x_refsource_MISC",
                            "x_cve_prog_secretariat_snapshot"
                        ],
                        "url": "https://github.com/aws/sagemaker-python-sdk/commit/2d873d53f708ea570fc2e2a6974f8c3097fe9df5/DEMO_2024_06_19"
                    },
                    {
                        "name": "https://github.com/aws/sagemaker-python-sdk/security/advisories/GHSA-7pc3-pr3q-58vg",
                        "tags": [
                            "x_refsource_CONFIRM",
                            "x_snapshot"
                        ],
                        "url": "https://github.com/aws/sagemaker-python-sdk/security/advisories/GHSA-7pc3-pr3q-58vg/adp-test-6-28"
                    },
                    {
                        "name": "https://github.com/aws/sagemaker-python-sdk/pull/4556",
                        "tags": [
                            "x_refsource_MISC",
                            "x_snapshot"
                        ],
                        "url": "https://github.com/aws/sagemaker-python-sdk/pull/4556/adp-test-6-28"
                    },
                    {
                        "name": "https://github.com/aws/sagemaker-python-sdk/commit/2d873d53f708ea570fc2e2a6974f8c3097fe9df5",
                        "tags": [
                            "x_refsource_MISC",
                            "x_snapshot"
                        ],
                        "url": "https://github.com/aws/sagemaker-python-sdk/commit/2d873d53f708ea570fc2e2a6974f8c3097fe9df5/adp-test-6-28"
                    }
                ]
            }
        ]
    }
}